Basic movement

Mashbar opened the pull request.

Mashbar committed:
Added new basic blocks

new sprite template added, starting to work on new player sprite

Mashbar committed:
8 way sprite

sprite draft for 8 directions, hooked it up to an animator, used a blend tree to handle the switching, basic script to point player in the direction they're moving

Jaffinch reviewed (changes requested):
It looks like the Tiled application has been accidentally included in this PR. Can you remove it and resubmit?

Mashbar committed:
Deleted Tiled

Removed Tiled from repo

Mashbar committed:
More Art

Updated player design

Mashbar committed:
Walking Animation

Art for animation and basic implementation into game, need better way of importing to unity, having to set custom pivot for each frame to compensate for isometric draw

Mashbar committed:
Raycast Controller

Done really rough maths for getting raycast points, 2 rays per point covers all directions, needs cleaning

Mashbar committed:
Raycast Controller

Very basic, very buggy, but it works... kinda

Mashbar committed:
New normal blocks

Decided raycasting in a diamond shape was silly and overly complicated, working on a new plan to handle the logic in a standard 2d plane and convert it to the isometric, so added assets to build boring square maps

Mashbar committed:
New normal raycast controller

basic controller with isometric mirror map linked, collisions are checked on normal map gameplay happens on the isometric

Mashbar committed:
RigidBody version

scrapped the last system, started using rigidbody physics, got good results quickly, movement is now mostly normalised (diags a little slower than verts), clipping fixed (moved the anchor to -0.9 on the y, don't know why it just works), Simple capsule collider positioned under the player gives correct motion within test box, all good stuff

Jaffinch committed:
Merge branch 'main' into pr/10
